Source/NSTimeZone.m

```diff
@@ -1330,22 +1330,23 @@ + (NSTimeZone*)systemTimeZone
         * Try to get timezone from windows system call.
         */
         {
-            TIME_ZONE_INFORMATION tz;
+            DYNAMIC_TIME_ZONE_INFORMATION tz;
             DWORD dst;
             wchar_t *tzName;
 
             // Get time zone name for US locale as expected by ICU method below
             LANGID origLangID = GetThreadUILanguage();
             SetThreadUILanguage(MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US));
-            dst = GetTimeZoneInformation(&tz);
+            dst = GetDynamicTimeZoneInformation(&tz);
             SetThreadUILanguage(origLangID);
```

Comment:
If the comments are to be believed, we can get rid of modifying the thread UI language altogether now that we are using the key name.
But we may want to roll that into the general timezone cleanup instead and leave it right now if it's not broke